tests/unit/render/test_render.cc

```diff
@@ -0,0 +1,199 @@
+  }
+
+  protected:
+    Render createRender(YAML::Node config, Info info, std::string &output_dir) {
+      // change output directory to use specific one for these tests
+      output_dir = "output/tests/render";
+
+      std::filesystem::path output_path(output_dir);
+      if (output_path.is_relative()) {
+        output_path = std::filesystem::path(SRC_DIR) / output_path;
+      }
+      output_dir = output_path.string();
+
+      // append / to avoid problems with file stems
+      if (!output_dir.empty() && output_dir.back() != '/') {
+        output_dir += '/';
+      }
+
+      return Render(
+      {
+          config["viz"]["rank_qoi"].as<std::string>(),
+          "",
+          config["viz"]["object_qoi"].as<std::string>()
+        },
+        config["viz"]["force_continuous_object_qoi"].as<bool>(),
+        info,
+        {
+          config["viz"]["x_ranks"].as<uint64_t>(),
+          config["viz"]["y_ranks"].as<uint64_t>(),
+          config["viz"]["z_ranks"].as<uint64_t>()
+        },
+        config["viz"]["object_jitter"].as<double>(),
+        output_dir,
+        config["output"]["file_stem"].as<std::string>(),
+        1.0,
+        config["viz"]["save_meshes"].as<bool>(),
+        // config["viz"]["save_pngs"]: Let PNG rendering to the ParseRenderTest test class.
+        false,
+        std::numeric_limits<PhaseType>::max()
+      );
+    }
+};
+
+/**
+ * Test Render:generate correcty run the different configuration files
+ */
+TEST_P(RenderTest, test_render_from_config) {
+  std::string const & config_file = GetParam();
+  YAML::Node config = YAML::LoadFile(fmt::format("{}/tests/config/{}", SRC_DIR, config_file));
+  Info info = Generator::loadInfoFromConfig(config);
+
+  uint64_t win_size = 2000;
+  uint64_t font_size = 50;
+
+  std::string output_dir;
+  if (config["viz"]["object_qoi"].as<std::string>() == "shared_block_id") {
+    // Temporary: this case must be removed as soon as the `shared_block_id` QOI becomes supported.
+    EXPECT_THROW(createRender(config, info, output_dir), std::runtime_error); // "Invalid Object QOI: shared_block_id"
+  } else {
+    // Render
+    Render render = createRender(config, info, output_dir);
+    render.generate(font_size, win_size);
+
+    // Check: that number of generated mesh files (*.vtp) correspond to the number of phases
+    std::string output_file_stem = config["output"]["file_stem"].as<std::string>();
+    for (uint64_t i = 0; i<info.getNumPhases(); i++) {
+      ASSERT_TRUE(std::filesystem::exists(fmt::format("{}{}_rank_mesh_{}.vtp", output_dir, output_file_stem, i))) << fmt::format("{}{}_rank_mesh_{}.vtp", output_dir, output_file_stem, i);
+      ASSERT_TRUE(std::filesystem::exists(fmt::format("{}{}_object_mesh_{}.vtp", output_dir, output_file_stem, i))) << fmt::format("{}{}_rank_mesh_{}.vtp", output_dir, output_file_stem, i);
+    }
+
+    // Validate mesh files (*.vtp) content ?
+    // Is it needed
```

Comment:
I think it is necessary--we want to make sure all output files are correct. @pierrepebay What do you think?

Reply:
Added to the follow up issue #95 because we have to know in terms of file content which specification we want to apply for these tests. A raw diff cannot be done because files are never the same. But some XML nodes presence or attributes might be tested.